src/main/java/Duke.java

```diff
@@ -1,10 +1,362 @@
+//                e.printStackTrace();
+//            }
+//        }
+    }
+
+    public void saving(LinkedList<Task> li){
+        try {
+            storage.writeFile(li);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+    
+    
+//    public static void main(String[] args) {
+//        new Duke("duke.txt");
+//    }
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    private Image user = new Image(this.getClass().getResourceAsStream("images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("images/DaDuke.png"));
+
+    /**
+     * Iteration 1:
+     * Creates a label with the specified text and adds it to the dialog container.
+     * @param text String containing text to add
+     * @return a label with the specified text that has word wrap enabled.
+     */
+    private Label getDialogLabel(String text) {
+        // You will need to import `javafx.scene.control.Label`.
+        Label textToAdd = new Label(text);
+        textToAdd.setWrapText(true);
+
+        return textToAdd;
+    }
+
+    /**
+     * Iteration 2:
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    private void handleUserInput() {
+        Label userText = new Label(userInput.getText());
+        Label dukeText = new Label(run(userInput.getText()));
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(userText.getText(), user),
+                DialogBox.getDukeDialog(dukeText.getText(), duke)
+
+        );
+        userInput.clear();
+    }
+
+    protected String getResponse(String input) {
+        return "Duke heard: " + input;
+    }
+
+    @Override
+    public void start(Stage stage) {
+        scrollPane = new ScrollPane();
+        dialogContainer = new VBox();
+        scrollPane.setContent(dialogContainer);
+
+        userInput = new TextField();
+        sendButton = new Button("Send");
+
+        AnchorPane mainLayout = new AnchorPane();
+        mainLayout.getChildren().addAll(scrollPane, userInput, sendButton);
+
+        scene = new Scene(mainLayout);
+
+        sendButton.setOnMouseClicked((event) -> {
+            String userWord = userInput.getText();
+            if (!userWord.isEmpty()) {
+                handleUserInput();
+            }
+        });
+
+        userInput.setOnAction((event) -> {
+            String userWord = userInput.getText();
+            if(!userWord.isEmpty()) {
+                handleUserInput();
+            }
+        });
+
+        dialogContainer.heightProperty().addListener((observable) -> scrollPane.setVvalue(1.0));
+
+        stage.setScene(scene);
+        stage.show();
     }
 }
```

Comment:
For run function, need to add abstract out the parsing to your parser class, it's too long.  Too many newline comments. 
Abstract out error handling, do not add to the main.